pkg/core/consensus/agreement/handler_test.go

```diff
@@ -29,6 +30,27 @@ func TestVoteVerification(t *testing.T) {
 	}
 }
 
+func TestGetVoterKeys(t *testing.T) {
+	p, keys := consensus.MockProvisioners(3)
+	hash, _ := crypto.RandEntropy(32)
+	ev := message.MockAgreement(hash, 1, 3, keys, p)
+	handler := NewHandler(keys[0], *p)
+
+	voterKeys := handler.getVoterKeys(ev)
+
+	// Ensure voterKeys only contains keys from `keys`
+	for _, key := range voterKeys {
+		found := false
+		for _, k := range keys {
+			if bytes.Equal(k.BLSPubKeyBytes, key) {
+				found = true
+			}
+		}
+
+		assert.True(t, found)
```

Comment:
use `require` instead of `assert`. Aliase it with assert at the beginning of the test